### DISCUSSION TITLE
The program based on Avalonia framework runs on raspberry pie

### DISCUSSION THREAD

**Discussion by rockRoll7** [AvaloniaUI/Avalonia#8207] [Q&A]
There is a fatal problem in the running of programs based on Avalonia framework on raspberry pie. An error will be reported when registering to run SIP services. Have you ever encountered it?

**Comment by rockRoll7** [comment]
Personal estimation is that there is a conflict between Avalonia framework and sip framework. How to solve it ?

**Comment by rockRoll7** [comment]
错误信息 : ../src/pj/os_core_unix.c:806: pj_thread_this: Assertion `!"Calling pjlib from unknown/external thread. You must " "register external threads with pj_thread_register() " "before calling any pjlib functions."' failed.

**Reply by kekekeks** [reply]
You probably need to register the thread with `pj_thread_register` before caling any pjlib functions, see https://www.pjsip.org/pjlib/docs/html/group__PJ__THREAD.htm#ga600d2f8baddfd78de3b6b60cce027a9a

**Reply by rockRoll7** [reply]
Thank you. It's like this. After the SIP service is registered in the Avalonia framework and runs for a period of time, there will be an error. However, there is no such problem in other frameworks

**Comment by rockRoll7** [comment]
Yes, after registering the SIP service in the Avalonia framework and running for a period of time, there will be an error. However, there is no such problem in other frameworks

**Reply by rockRoll7** [reply]
After registering the SIP service, the software reported an error after running for a period of time, and has not yet made a call

**Comment by rockRoll7** [comment]
How to solve this problem?
